docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises.md

```diff
@@ -750,6 +750,130 @@ As an alternative to setting crontab permissions, you can install systemd-specif
+
+   ```output
+   * Starting YB Node Agent install
+   * Creating Node Agent Directory
+   * Changing directory to node agent
+   * Creating Sub Directories
+   * Downloading YB Node Agent build package
+   * Getting Linux/amd64 package
+   * Downloaded Version - 2.17.1.0-PRE_RELEASE
+   * Extracting the build package
+   * The current value of Node IP is not set; Enter new value or enter to skip: 10.9.198.2
+   * The current value of Node Name is not set; Enter new value or enter to skip: Test
+   * Select your Onprem Provider
+   1. Provider ID: 41ac964d-1db2-413e-a517-2a8d840ff5cd, Provider Name: onprem
+           Enter the option number: 1
+   * Select your Instance Type
+   1. Instance Code: c5.large
+           Enter the option number: 1
+   * Select your Region
+   1. Region ID: dc0298f6-21bf-4f90-b061-9c81ed30f79f, Region Code: us-west-2
+           Enter the option number: 1
+   * Select your Zone
+   1. Zone ID: 99c66b32-deb4-49be-85f9-c3ef3a6e04bc, Zone Name: us-west-2c
+           Enter the option number: 1
+           • Completed Node Agent Configuration
+           • Node Agent Registration Successful
+   You can install a systemd service on linux machines by running node-agent-installer.sh -t install-service (Requires sudo access).
+   ```
+
+4. Run the following command to enable the node agent as a systemd service, which is required for self-upgrade and other functions: 
+
+   ```sh
+   sudo node-agent-installer.sh -t install-service  
+   ```
+
+
+### Registration
+
+To enable secured communication, the node agent is automatically registered during its installation so the YugabyteDB Anywhere is aware of its existence. You can also register and unregister the node agent manually during configuration.
+
+The YugabyteDB Anywhere API provides the endpoint via the following API: 
+
+```
+POST /api/v1/customers/<customer_id>/node_agents
+```
+
+The following is the node agent registration command:
+
+```sh
+node-agent node register --api-token <api_token>
+```
+
+If you need to overwrite any previously configured values, you can use the following parameters within the registration command:
+
+- `--node_ip` represents the node IP address.
+- `--url` represents the YugabyteDB Anywhere address.
+
+For secured communication, YugabyteDB Anywhere generates a key pair (private, public, and server certificate) that is sent to the node agent as part of its registration process.
+
+You can obtain a list of existing node agents using the following API: 
+
+```http
+GET /api/v1/customers/<customer_id>/node_agents
+```
+
+To unregister a node agent, use the following API: 
+
+```http
+DELETE /api/v1/customers/<customer_id>/node_agents/<node_agent_id>
+```
+
+### Operations
+
+A node agent supports the following commands:
+
+- `node-agent node register` is used for registering a node and node agent to YugabyteDB Anywhere. This is needed before starting a node agent server. YugabyteDB Anywhere can maintain a table for node agents.
```

Comment:
`node-agent node register` is used for registering a node and node agent to YugabyteDB Anywhere if it was unregistered manually. Registering an already registered node agent fails as YugabyteDB Anywhere keeps a record of the node agent with this IP.

Reply:
Changed